azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -271,10 +273,15 @@ public void setGlobalProps(Props globalProps) {
     // Every 2 mins clean the recently finished list
     private static final long RECENTLY_FINISHED_INTERVAL_MS = 2 * 60 * 1000;
 
+    // Every 5 mins kill flows running longer than allowed max running time
+    private static final long LONG_RUNNING_FLOW_KILLING_INTERVAL_MS = 5 * 60 * 1000;
```

Comment:
TL DR; There are some advantages to using classes like `Duration` as opposed to using primitives like `int` and `long` 

@chengren311 @HappyRay 
I totally understand the point you are trying to make but let me share a story. :)

This was narrated to me by Prof. Stroustrup himself. It's about a NASA probe which suffered a horrible crash probably more than a decade ago and costed more than $100 M. The controller code was written in C++. On investigation they found out that there were 2 components reading the same `double` constant which had a major role to play in the crash. I don't exactly remember the details but one of them was developed by the Americans and the other by some European country. The crash happened because the retro rockets which were supposed to stabilize and slowly land the probe stopped firing when the probe was 100 ft in the air. It was not a happy landing. On digging through the logs they found out the problem. As it turns out the Americans are fond of the pound system and the Europeans prefer the metric system. There was this evil `double` value which missed all conversions; slipped through the compiler and through code reviews and was passed as is to another component which expected a `double` but in different units. Rest is history :)

Here we have Azkaban! which is open source :D However, I would still prefer using a dedicated data structure for measuring time like `Duration` as opposed to using `long` or `int`. While this convention has been there for decades in C and other languages, it hasn't always yielded the best results.

Regarding intuition, `Duration.ofMinutes(5)` is more intuitive to me than a `5*60*1000` long with a comment. It is also very safe since the numbers are abstracted away in the API itself. The joda-time library has been extremely popular in the Java community and it was officially incorporated into Java 8 itself because of its simplicity and utility. Now, since we have Java 8 already, we can directly benefit from it instead of having variable *names* determine the *units*.